### DISCUSSION TITLE
Не работает голосовой чат в Discord

### DISCUSSION THREAD

**Discussion by avel07** [Anonym-tsk/nfqws-keenetic#85] [Обсуждения]
Не работают голосовые чаты в Discord. 
https://github.com/bol-van/zapret/issues/455 - тут описана стратегия с --dpi-desync-any-protocol=1.
Но это не приводит к успеху, всё такие же проблемы)
У кого-нибудь получилось запустить голосовой чат в дискорде?

**Comment by vovan190310** [comment]
> Не работают голосовые чаты в Discord. [bol-van/zapret#455](https://github.com/bol-van/zapret/issues/455) - тут описана стратегия с --dpi-desync-any-protocol=1. Но это не приводит к успеху, всё такие же проблемы) У кого-нибудь получилось запустить голосовой чат в дискорде?

получилось,я запустил впн,открыл приложение дискорд,открыл гудбайдпи потом вырубил впн и всё ок работает

**Comment by avel07** [comment]
≈@vovan190310 Я слышу собеседника, но он меня не слышит) 
Да и в целом хочется нормальное практичное решение в самом роутере)

**Comment by vovan190310** [comment]
> ≈@vovan190310 Я слышу собеседника, но он меня не слышит) Да и в целом хочется нормальное практичное решение в самом роутере)

извини,тут не могу помочь. Просто дал простой совет:)

**Comment by dartraiden** [comment]
> Но это не приводит к успеху

Вы же в `NFQWS_ARGS_QUIC` это пытались добавлять, правда?

Голосовые чаты у дискорда бегают именно через QUIC

**Comment by avel07** [comment]
в файле nfqws.conf 
К дефолтному конфигу просто добавил --dpi-desync-any-protocol=1
```
# QUIC strategy
NFQWS_ARGS_QUIC="--dpi-desync=fake --dpi-desync-any-protocol=1 --dpi-desync-repeats=6 --dpi-desync-cutoff=d4 --dpi-desync-fooling=badsum --dpi-desync-fake-quic=/opt/etc/nfqws/quic_initial.bin"
```

**Comment by XE1A3R** [comment]
> в файле nfqws.conf К дефолтному конфигу просто добавлен --dpi-desync-any-protocol=1
> 
> ```
> # QUIC strategy
> NFQWS_ARGS_QUIC="--dpi-desync=fake --dpi-desync-any-protocol=1 --dpi-desync-repeats=6 --dpi-desync-cutoff=d4 --dpi-desync-fooling=badsum --dpi-desync-fake-quic=/opt/etc/nfqws/quic_initial.bin"
> ```

Тоже пробовал, пока что без успешно.

**Comment by lev1n-ap** [comment]
Я просто не парился, подключился через бота в тг - https://t.me/flexiblevpn_bot
И голосовые чаты и всё работает))

**Comment by avel07** [comment]
Ну тогда проще на роутере начать назворачивать AmnesiaWG конфигурацию какую-нибудь. Например через Wireguard.
Как-нибудь бы без VPN с подменой пакетов. Проверял на windows пакетик один с zapret в коробке. Работало. Но у меня рабочий macOS да и телефон с IOS. Хотелось бы всё же пакеты подменять.

**Comment by dartraiden** [comment]
> К дефолтному конфигу просто добавил --dpi-desync-any-protocol=1

1) [Достаточно](https://ntc.party/t/zapret-%D0%BE%D0%B1%D1%81%D1%83%D0%B6%D0%B4%D0%B5%D0%BD%D0%B8%D0%B5/726/1119) `--dpi-desync-any-protocol`

2) См. https://ntc.party/t/%D0%BE%D0%B1%D1%81%D1%83%D0%B6%D0%B4%D0%B5%D0%BD%D0%B8%D0%B5-%D0%BE%D0%B1%D1%85%D0%BE%D0%B4-%D0%B7%D0%B0%D0%BC%D0%B5%D0%B4%D0%BB%D0%B5%D0%BD%D0%B8%D1%8F-youtube-%D0%B8-%D0%B1%D0%BB%D0%BE%D0%BA%D0%B8%D1%80%D0%BE%D0%B2%D0%BE%D0%BA-%D0%BD%D0%B0-%D0%BF%D1%80%D0%BE%D1%88%D0%B8%D0%B2%D0%BA%D0%B5-keenetic-%D1%81-%D0%BF%D0%BE%D0%BC%D0%BE%D1%89%D1%8C%D1%8E-nfqws-tpws-zapret/9426/228

**Comment by avel07** [comment]
@dartraiden Именно так я и делал. В файле /etc/auto.list добавил список доменов:
`discord.gg
discord.com
discordapp.com
discord.media
discordapp.net
discordcdn.com
discord.dev
discord.new
discord.gift
discordstatus.com
dis.gd
discord.co
discordsays.com
discord.design
discord.gifts
discord.store
discord.tools
discordmerch.com
discordpartygames.com
discord-activities.com
discordactivities.com
zendesk.com`

Далее в файле nfqws.conf добавил ``` --dpi-desync-any-protocol.```
Далее в файле S51nqqws вставил как вы указали в ссылке порт 65535 - 
```RULE_QUIC="-t mangle -p udp --dport 443:65535 -m connbytes --connbytes-dir=original --connbytes-mode=packets --connbytes 1:8 -m mark ! --mark 0x40000000/0x40000000 -j NFQUEUE --queue-num $NFQUEUE_NUM_QUIC --queue-bypass"```

Но всё равно не работает войс =)

**Comment by CREAsTIVE** [comment]
Для меня заработало:
```
start "zapret: http,https,quic" /min "%~dp0winws.exe" ^
--wf-tcp=443-65535 --wf-udp=443-65535 ^
--filter-udp=50000-65535 --dpi-desync=fake,tamper --dpi-desync-any-protocol --dpi-desync-fake-quic="%~dp0quic_initial_www_google_com.bin" --new
```
как дополнение к уже существующему кофигу.

**Comment by avel07** [comment]
@CREAsTIVE это же экзешник с zapret. 
А как в Keenetic сделать?)

**Comment by avel07** [comment]
Решено с обновлением.

**Comment by Q0011er** [comment]
Обновился с версии 1 от начала августа 2024 на текущую 2.4.1 ютюб на телеке samsung работает, дискорд и голос в нем работают.

**Comment by ve1mi** [comment]
Тоже не работал rtc в дискорде (было бесконечное подключение) пока не добавил в auto.list дополнительные домены. Сам не знаю какой из доменов помог подключиться головому чату, но вычислять не хочется 
Настройки nfqws стоковые. Версия 2.7.2
```
discordapp.net
discordapp.com
discord.gg
images-ext-1.discordapp.net
www.discord.com
www.discord.app
discord.app
discord.media
discordcdn.com
discord.dev
discord.new
discord.gift
discordstatus.com
dis.gd
dl.discordapp.net
discordapp.io
updates.discord.com
```

**Reply by Karolina2a** [reply]
Подскажи, добрый человек, человеку который абсолютно не шарит, где найти файл в который это нужно вставить 🙏🙏🙏

**Reply by azat-b** [reply]
> Подскажи, добрый человек, человеку который абсолютно не шарит, где найти файл в который это нужно вставить 🙏🙏🙏

Добрый день! Необходимо зайти по адресу роутера (192.168.0.1, или иное) с указанием порта :90. Далее во вкладку auto.list это добавить.

**Comment by smertnickx** [comment]
Ну, во-первых RTC дискорда ходит не через QUIC - через него гоняются стрим потоки видео YT
Войс ходит через UDP, соответственно крутить надо его конфиги. в моем случае работает так.
```
NFQWS_ARGS_UDP="--filter-udp=50000-50099 --filter-l7=discord,stun --dpi-desync=fake --dpi-desync-repeats=6"
```
Но на старых версиях NFQWS может не работать, лучше сначала обновиться до актуальной, данный конфиг в ней есть., проверял на 2.8.4 и на текущей

**Comment by sergeykupreenkov** [comment]
У кого-то получилось завести дискорд? У меня на 2.8.6 звонки не заработали

**Reply by Karolina2a** [reply]
> У кого-то получилось завести дискорд? У меня на 2.8.6 звонки не заработали

Попробуй команду  -Ku -a3 -An -Kt,h -s1 -q1 -Y -Ar но с ней не будет Ютуб. Но для Ютуба легче найти код, чем для дискорда, поэтому немдумаю, что это проблема)
